### PR TITLE
Rename ESP32 to XTensaLX6 and split crate into rt and base

### PR DESCRIPTION
**Pull request by arjanmels**
This PR renames ESP32 (SOC name) to XtensaLX6 (CPU name) to align to convention of other platforms.

It also adds seperate xtensa-lx6-rt and xtensa-lx6 support (instead of just xtensa-lx6-rt).

### CODE REVIEW

File: ci/script.sh [hunk 1]
```diff
@@ -697,10 +697,13 @@ main() {
             echo '[dependencies.bare-metal]' >> $td/Cargo.toml
             echo 'version = "0.2.0"' >> $td/Cargo.toml
 
+            echo '[dependencies.xtensa-lx6]' >> $td/Cargo.toml
+            echo 'git = "https://github.com/esp-rs/xtensa-lx6.git"' >> $td/Cargo.toml
```
Comment (MabezDev) [inline ci/script.sh:701]:
```suggestion
            echo 'version = "0.1.0"' >> $td/Cargo.toml
```

File: ci/script.sh [hunk 2]
```diff
@@ -697,10 +697,13 @@ main() {
...
+
             echo '[dependencies.xtensa-lx6-rt]' >> $td/Cargo.toml
             echo 'git = "https://github.com/esp-rs/xtensa-lx6-rt.git"' >> $td/Cargo.toml
```
Comment (MabezDev) [inline ci/script.sh:704]:
```suggestion
            echo 'version = "0.2.0"' >> $td/Cargo.toml
```

**Comment by burrbull** [comment]
cc @MabezDev

**Comment by arjanmels** [comment]
@burrbull Thanks for the speedy review and merge.

**Pull request merged**